### PR TITLE
Update drbg.c

### PR DESCRIPTION
**Pull request by fochoao-alt**
##### Checklist

- [x] Fix of more entropy

### CODE REVIEW

**Comment by andrewkdinh** [comment]
What does this PR attempt to accomplish and why should it be merged? Please update the description

**Comment by fochoao-alt** [comment]
> What does this PR attempt to accomplish and why should it be merged? Please update the description

See the title again, more entropy.

**Comment by nhorman** [comment]
I'm sorry @fochoao-alt , I don't quite buy that.  Can you please explain how this gathers additional entropy?  This doesn't even look like it will build to me.

Additionally, could you please sign and submit a cla so that we can properly review this?   Instructions can be found [here](https://openssl-library.org/policies/cla/)

File: providers/implementations/rands/drbg.c [hunk 1]
```diff
@@ -446,18 +446,27 @@ int ossl_prov_drbg_instantiate(PROV_DRBG *drbg, unsigned int strength,
 
     if (!drbg->instantiate(drbg, entropy, entropylen, nonce, noncelen,
                            pers, perslen)) {
-        cleanup_entropy(drbg, entropy, entropylen);
         ERR_raise(ERR_LIB_PROV, PROV_R_ERROR_INSTANTIATING_DRBG);
         goto end;
     }
-    cleanup_entropy(drbg, entropy, entropylen);
-
+    l_prov_drbg_instantiate(PROV_DRBG *drbg, unsigned int strength,
```
Comment (nhorman) [inline providers/implementations/rands/drbg.c:452]: This isn't right, an unclosed call to l_prov_drbg_instatiate?  This looks more like a function prototype than a call.

**Review by nhorman — CHANGES_REQUESTED** [review]

File: providers/implementations/rands/drbg.c [hunk 2]
```diff
@@ -513,7 +522,7 @@ static int ossl_prov_drbg_reseed_unlocked(PROV_DRBG *drbg,
         }
         if (ent_len > drbg->max_entropylen) {
             ERR_raise(ERR_LIB_RAND, RAND_R_ENTROPY_INPUT_TOO_LONG);
-            drbg->state = EVP_RAND_STATE_ERROR;
+            drbg->state = EVP_RAND_STATE_ERROR;int oss
```
Comment (nhorman) [inline providers/implementations/rands/drbg.c:525]: This seems like you left a trailing variable declaration in the code.

**Review by nhorman — CHANGES_REQUESTED** [review]

**Review by fochoao-alt — COMMENTED** [review]
Fixed it.

File: providers/implementations/rands/drbg.c (hunk 2 above)
Comment (fochoao-alt) [reply providers/implementations/rands/drbg.c]: Fixed, but I cannot sign the document until Monday.

**Comment by fochoao-alt** [comment]
> I'm sorry @fochoao-alt , I don't quite buy that. Can you please explain how this gathers additional entropy? This doesn't even look like it will build to me.
> 
> Additionally, could you please sign and submit a cla so that we can properly review this? Instructions can be found [here](https://openssl-library.org/policies/cla/)

Fixed the code, thanks for pointing that out as well... but cannot sign the document until Monday next week.

**Review by fochoao-alt — COMMENTED** [review]
For review.

**Comment by nhorman** [comment]
no, this really isn't fixed yet (see the ci run), nor have you explained what exactly this patch is doing to "obtain more entropy"

**Comment by fochoao-alt** [comment]
I see it failing horribly, will wait for Monday, and fix it in full.

**Comment by fochoao-alt** [comment]
> no, this really isn't fixed yet (see the ci run), nor have you explained what exactly this patch is doing to "obtain more entropy"

Thanks, yes, wait for Monday then, I will have the code by then as well.

**Review by paulidale — COMMENTED** [review]
It is not at all clear what this is trying to do.  As it stands it cannot be accepted.

The generate call is essential.  As is the enable locking call.  You've removed both.

**Comment by t8m** [comment]
This looks like some AI generated patches. Closing.

**Pull request closed without merging**